### DISCUSSION TITLE
GLSP client production build

### DISCUSSION THREAD

**Discussion by sanjeev55** [eclipse-glsp/glsp#989] [Q&A]
Hello,

Can anyone help with correct docker configuration to build a production image of the glsp client project. I have tried the following in the docker file:

```
RUN yarn install --production && \
    yarn cache clean
```

This builds the image with significantly reduced size but does not run properly (or at all). I have following dependencies and devDependencies:

```
"dependencies": {
    "@eclipse-glsp/client": "^1.0.0",
    "ffmpeg": "0.0.4",
    "node-sass": "^8.0.0",
    "theia": "^2.1.2"
  },

"devDependencies": {
    "@eclipse-glsp/config": "^1.0.0",
    "lerna": "^4.0.0",
    "typescript": "~4.5.5"
  },
``` 

Without the production flag i.e:

```
 RUN yarn && \
     NODE_OPTIONS="--max_old_space_size=4096"

```

The size of the image increases and builds properly and runs as expected. I noticed that with the production flag, the individual `workspaces` in the project doesn't build. I have something like this in package.json:

```
"workspaces": [
    "app-glsp",
    "app-theia",
    "browser-app",
    "theia-custom-extensions"
  ],
```
Is it because of  `lerna` not being included in one of the devDependencies? Or any other reasons? Any help on this would be highly appreciated. 
Thanks.

**Comment by tortmayr** [comment]
Hi @sanjeev55,
unfortunately I don't have much experience with production builds. At first glance your approach looks correct. Are there any errors logged if your run the production image? If so, could you please post the log?
@planger @martin-fleck-at Do you have an idea what`s going wrong here?

**Reply by planger** [reply]
No unfortunately I don't have a good idea. :-/
